### PR TITLE
Remove string allocation in the count of subscribers and publishers

### PR DESCRIPTION
**Pull request by deng02**
Addresses https://github.com/ros2/rmw_fastrtps/issues/166

### CODE REVIEW

File: rmw_fastrtps_cpp/src/rmw_count.cpp [hunk 1]
```diff
@@ -48,34 +48,19 @@ rmw_count_publishers(
 
   CustomParticipantInfo * impl = static_cast<CustomParticipantInfo *>(node->data);
 
-  std::map<std::string, std::vector<std::string>> unfiltered_topics;
```
Comment (clalancette) [inline rmw_fastrtps_cpp/src/rmw_count.cpp]: I think you need to set `*count` to 0 here (like you do in `rmw_count_subscribers`), no?

Comment (deng02) [reply rmw_fastrtps_cpp/src/rmw_count.cpp]: Yes, I'll fix this.

Comment (deng02) [reply rmw_fastrtps_cpp/src/rmw_count.cpp]: PR Updated

File: rmw_fastrtps_cpp/src/rmw_count.cpp [hunk 2]
```diff
@@ -48,34 +48,20 @@ rmw_count_publishers(
...
     "rmw_fastrtps_cpp",
     "looking for subscriber topic: %s", topic_name)
-  for (auto it : unfiltered_topics) {
-    RCUTILS_LOG_DEBUG_NAMED(
-      "rmw_fastrtps_cpp",
-      "available topic: %s", it.first.c_str())
-  }
   RCUTILS_LOG_DEBUG_NAMED(
     "rmw_fastrtps_cpp",
     "number of matches: %zu", *count)
```
Comment (dirk-thomas) [inline rmw_fastrtps_cpp/src/rmw_count.cpp]: Should those two log messages be merged into one now?

Same below.

Comment (deng02) [reply rmw_fastrtps_cpp/src/rmw_count.cpp]: Agreed.  I'll collapse this.

Comment (deng02) [reply rmw_fastrtps_cpp/src/rmw_count.cpp]: PR updated

**Comment by dirk-thomas** [comment]
Thank you for the patch and the quick updates.

**Pull request merged**